frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -27,59 +27,66 @@ import { ProvisionSource } from '../../../../constants/vm/provision-source';
 import { prefillVmTemplateUpdater } from './prefill-vm-template-state-update';
 import { iGetPrameterValue, iGetAnnotation } from '../../../../selectors/immutable/common';
 import { CDI_UPLOAD_POD_ANNOTATION, CDI_UPLOAD_RUNNING } from '../../../cdi-upload-provider/consts';
+import { iGetIn } from '../../../../utils/immutable';
 
 const selectUserTemplateOnLoadedUpdater = (options: UpdateOptions) => {
   const { id, dispatch, getState } = options;
   const state = getState();
 
   if (
-    iGetVmSettingAttribute(state, id, VMSettingsField.USER_TEMPLATE, 'initialized') ||
-    !options.changedCommonData.has(VMWizardProps.userTemplates)
+    iGetCommonData(state, id, VMWizardProps.isUserTemplateInitialized) ||
+    !iGetLoadedCommonData(state, id, VMWizardProps.userTemplate)
   ) {
     return;
   }
 
-  const userTemplateName = iGetCommonData(state, id, VMWizardProps.userTemplateName);
-  if (!userTemplateName) {
-    return;
-  }
-
-  const userTemplates = iGetLoadedCommonData(state, id, VMWizardProps.userTemplates);
-  const isUserTemplateValid = userTemplates?.find(
-    (template) => iGetName(template) === userTemplateName,
+  dispatch(
+    vmWizardInternalActions[InternalActionType.UpdateCommonDataValue](
+      id,
+      [VMWizardProps.isUserTemplateInitialized],
+      true,
+    ),
   );
+};
+
+const userTemplateValidityCheck = (options: UpdateOptions) => {
+  const { id, dispatch, getState } = options;
+  const state = getState();
+
+  const userTemplate = iGetCommonData(state, id, VMWizardProps.userTemplate);
 
-  if (!isUserTemplateValid) {
+  if (!userTemplate) {
     return;
   }
 
-  dispatch(
-    vmWizardInternalActions[InternalActionType.UpdateVmSettings](id, {
-      [VMSettingsField.USER_TEMPLATE]: {
-        initialized: true,
-        value: userTemplateName,
-      },
-    }),
-  );
+  const isLoaded = iGetIn(userTemplate, ['loaded']);
+  const isLoadError = iGetIn(userTemplate, ['loadError']);
+
+  if (!isLoaded && isLoadError) {
+    const isDisabled = asDisabled(true, VMWizardProps.userTemplate);
+
+    const disableAllVMSettingsFields = Object.keys(VMSettingsField).reduce((acc, key) => {
+      return { [key]: { isDisabled }, ...acc };
+    }, {});
+
+    dispatch(
+      vmWizardInternalActions[InternalActionType.UpdateVmSettings](id, {
+        ...disableAllVMSettingsFields,
+      }),
+    );
+  }
 };
 
 const selectedUserTemplateUpdater = (options: UpdateOptions) => {
   const { id, prevState, dispatch, getState } = options;
   const state = getState();
-  if (!hasVmSettingsChanged(prevState, state, id, VMSettingsField.USER_TEMPLATE)) {
+
+  if (!hasCommonDataChanged(prevState, state, id, VMWizardProps.isUserTemplateInitialized)) {
```

Comment:
move it or copy it ? if we move it what should be here instead ?

Reply:
two reasons:
1. we don't want to disable the fields on each update if they already disabled.
2. we don't want to execute `prefillVmTemplateUpdater` on every update - the user would not be able to make changes at the flavor field for example

Reply:
I wasn't able to move the content of this function to `selectUserTemplateOnLoadedUpdater` the fields are not getting populated. if you have a suggestion please share it, otherwise let's do it as a followup. wdyt ?

Reply:
I don't see any issue with it. It seems pretty simple:

```ts
const selectUserTemplateOnLoadedUpdater = (options: UpdateOptions) => {
  const { id, dispatch, getState } = options;
  const state = getState();

  if (
    iGetCommonData(state, id, VMWizardProps.isUserTemplateInitialized) ||
    !iGetLoadedCommonData(state, id, VMWizardProps.userTemplate)
  ) {
    return;
  }

  dispatch(
    vmWizardInternalActions[InternalActionType.UpdateCommonDataValue](
      id,
      [VMWizardProps.isUserTemplateInitialized],
      true,
    ),
  );

  const iUserTemplate = iGetLoadedCommonData(state, id, VMWizardProps.userTemplate);
  const isDisabled = asDisabled(iUserTemplate != null, VMWizardProps.userTemplate);

  dispatch(
    vmWizardInternalActions[InternalActionType.UpdateVmSettings](id, {
      [VMSettingsField.PROVISION_SOURCE_TYPE]: { isDisabled },
      [VMSettingsField.CONTAINER_IMAGE]: { isDisabled },
      [VMSettingsField.IMAGE_URL]: { isDisabled },
      [VMSettingsField.OPERATING_SYSTEM]: { isDisabled },
      [VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE]: {
        isHidden: asHidden(iUserTemplate != null, VMWizardProps.userTemplate),
      },
    }),
  );

  prefillVmTemplateUpdater(options);
};
```

Reply:
This is the first thing I tried. it doesn't work.